File: backend/app/schemas/contract.py
from datetime import date, datetime
from typing import List, Optional
from pydantic import BaseModel, Field


class ContractBase(BaseModel):
    contract_number: str
    title: str
    description: Optional[str] = None
    department: str
    contractor: Optional[str] = None
    baseline_value: Optional[float] = None
    current_value: Optional[float] = None
    baseline_start_date: Optional[date] = None
    current_start_date: Optional[date] = None
    baseline_completion_date: Optional[date] = None
    current_completion_date: Optional[date] = None
    status: Optional[str] = "ACTIVE"


class ContractCreate(ContractBase):
    pass


class ContractUpdate(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    department: Optional[str] = None
    contractor: Optional[str] = None
    baseline_value: Optional[float] = None
    current_value: Optional[float] = None
    baseline_start_date: Optional[date] = None
    current_start_date: Optional[date] = None
    baseline_completion_date: Optional[date] = None
    current_completion_date: Optional[date] = None
    status: Optional[str] = None
    risk_score: Optional[float] = None
    risk_level: Optional[str] = None


class ContractResponse(ContractBase):
    id: str
    risk_score: Optional[float] = None
    risk_level: Optional[str] = None
    created_by: Optional[str] = None
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class ContractSummaryResponse(BaseModel):
    id: str
    contract_number: str
    title: str
    department: str
    contractor: Optional[str] = None
    status: str
    risk_score: Optional[float] = None
    risk_level: Optional[str] = None
    
    baseline_value: Optional[float] = None
    current_value: Optional[float] = None
    cost_drift_percentage: Optional[float] = None
    
    baseline_completion_date: Optional[date] = None
    current_completion_date: Optional[date] = None
    schedule_drift_months: Optional[float] = None
    
    version_count: int = 0
    document_count: int = 0
    change_count: int = 0
    latest_review_decision: Optional[str] = None
    last_updated: datetime

File: frontend/src/types/contract.ts
export type RiskLevel = 'LOW' | 'MEDIUM' | 'HIGH' | 'CRITICAL';

export type ContractStatus = 'ACTIVE' | 'UNDER_REVIEW' | 'CLEARED' | 'ESCALATED' | 'CLOSED';

export interface Contract {
  id: string;
  contractNumber: string;
  contract_number?: string;
  title: string;
  description?: string | null;
  department: string;
  contractor?: string | null;
  baselineValue: number | null;
  baseline_value?: number | null;
  currentValue: number | null;
  current_value?: number | null;
  baselineStartDate?: string | null;
  baseline_start_date?: string | null;
  currentStartDate?: string | null;
  current_start_date?: string | null;
  baselineCompletionDate?: string | null;
  baseline_completion_date?: string | null;
  currentCompletionDate?: string | null;
  current_completion_date?: string | null;
  riskScore: number | null;
  risk_score?: number | null;
  riskLevel: RiskLevel | null;
  risk_level?: RiskLevel | null;
  status: ContractStatus;
  createdBy?: string | null;
  created_by?: string | null;
  createdAt: string;
  created_at?: string;
  updatedAt: string;
  updated_at?: string;
}

export interface ContractSummary {
  id: string;
  contract_number: string;
  title: string;
  department: string;
  contractor: string | null;
  status: ContractStatus;
  risk_score: number | null;
  risk_level: RiskLevel | null;
  baseline_value: number | null;
  current_value: number | null;
  cost_drift_percentage: number | null;
  baseline_completion_date: string | null;
  current_completion_date: string | null;
  schedule_drift_months: number | null;
  version_count: number;
  document_count: number;
  change_count: number;
  latest_review_decision: string | null;
  last_updated: string;
}

export interface PaginatedContracts {
  items: Contract[];
  total: number;
  page: number;
  page_size: number;
  total_pages: number;
}
